File: level_2/타겟 넘버.py
answer = 0
def track(depth, depth_limit, cur_sum, target):
    global answer
    if depth == depth_limit:
        if cur_sum == target:
            answer += 1
        return

    for i in range(2):
        if i == 0:
            cur_sum += numbers[depth]
            track(depth + 1, depth_limit, cur_sum, target)
            cur_sum -= numbers[depth]
        else:
            cur_sum -= numbers[depth]
            track(depth + 1, depth_limit, cur_sum, target)
            cur_sum += numbers[depth]

def solution(numbers, target):
    track(0, len(numbers), 0, target)


numbers = [1,1,1,1,1]
target = 3
track(0,len(numbers),0,target)

print(answer)

File: Java/src/BFS_DFS/타겟넘버.java
package BFS_DFS;

public class 타겟넘버 {
	int count = 0;
    
    public int solution(int[] numbers, int target) {
        track(0,0,numbers,target);
        return count;
    }
    
    void track(int depth, int curr_sum, int[] numbers, int target) {
        if(depth == numbers.length) {
            if(curr_sum == target) {
                count++;
            }
            return;
        }       
        track(depth+1, curr_sum + numbers[depth],numbers, target);
        track(depth+1, curr_sum - numbers[depth],numbers, target);
    }
}
